pkg/generate/app/cmd/newapp.go

```diff
@@ -717,13 +717,86 @@ func (c *AppConfig) checkCircularReferences(objects app.Objects) error {
 	for _, obj := range objects {
 		if bc, ok := obj.(*buildapi.BuildConfig); ok {
 			input := buildutil.GetInputReference(bc.Spec.Strategy)
-			if bc.Spec.Output.To != nil && input != nil &&
-				reflect.DeepEqual(input, bc.Spec.Output.To) {
-				ns := input.Namespace
-				if len(ns) == 0 {
-					ns = c.OriginNamespace
+			output := bc.Spec.Output.To
+
+			if output == nil || input == nil {
+				return nil
+			}
+
+			inNS := input.Namespace
+			if len(inNS) == 0 {
+				inNS = c.OriginNamespace
+			}
+
+			// We might be outputing a docker image that would trigger our imagestream repo.
```

Comment:
> That's it. I'm writing a spell checker for vscode :)

For Atom it is by default disabled for Go files... you need to somehow enable it for only the comments. Might be the same with VScode.